### PR TITLE
First row is global across cali-query processes

### PR DESCRIPTION
**Pull request by alfredo-gimenez**
Fixes another problem in #67

### CODE REVIEW

**Comment by daboehme** [comment]
We'll eventually need to add some unit tests for the JsonFormatter

File: src/reader/JsonFormatter.cpp [hunk 1]
```diff
@@ -59,6 +59,8 @@ const std::string opt_split = std::string("split");
 const std::string opt_pretty = std::string("pretty");
 const std::string opt_quote_all = std::string("quote-all");
 
+bool g_first_row = true;
```
Comment (daboehme) [inline src/reader/JsonFormatter.cpp]: You need to reset this in JsonFormatter::flush() @alfredo-gimenez

Comment (daboehme) [reply src/reader/JsonFormatter.cpp]: And make it an atomic while you're at it

Comment (daboehme) [reply src/reader/JsonFormatter.cpp]: In fact, was the problem just a race condition? I don't think this has to be a global variable, just needs to be protected between threads.

**Comment by DavidPoliakoff** [comment]
Yeah, we're def still getting the same results :(

**Comment by daboehme** [comment]
@DavidPoliakoff: As a workaround try `--threads 1` in cali-query

**Comment by DavidPoliakoff** [comment]
@daboehme : appreciated, but I have a much dumber workaround already implemented in my Python scripts, will keep it in mind though

File: src/reader/JsonFormatter.cpp (hunk 1 above)
Comment (alfredo-gimenez) [reply src/reader/JsonFormatter.cpp]: just saw this, yea it was a race condition. I previously thought every thread was getting it's own `m_first_row`, is that not the case then? I can change it back to class scope if so, it's now thread-safe on the latest.

**Comment by DavidPoliakoff** [comment]
Latest push from @alfredo-gimenez fixed my issues, LGTM

File: src/reader/JsonFormatter.cpp (hunk 1 above)
Comment (daboehme) [reply src/reader/JsonFormatter.cpp]: There is only one formatter instance for all threads. However, you can create multiple formatters over time if you do multiple flushes. Better put it back in the class scope.

Comment (alfredo-gimenez) [reply src/reader/JsonFormatter.cpp]: sounds good, done.

**Pull request merged**